### PR TITLE
Added functions overview

### PR DESCRIPTION
**Pull request by cevou**
Added an overview of all implemented functions and how they are supported by the different engnes. Also added simplify, maxDistance and snapToGrid functions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -120,3 +120,8 @@ Usage
 -----
 
 To be written.
+
+Functions
+---------
+
+All suported spacial functions can be found in the [functions.md](functions.md) file.
```
Comment (BenMorel) [inline README.md]: Typo: spa**t**ial

Comment (BenMorel) [inline README.md]: IMO the function list is not that big, and would be better suited in the README directly?

File: src/Geometry.php [hunk 2]
```diff
@@ -526,7 +526,49 @@ public function symDifference(Geometry $geometry)
     }
 
     /**
-     * Returns the raw coordinates of this Goemetry as an array.
+     * Snap all points of the input geometry to a regular grid.
+     *
+     * @noproxy
+     *
+     * @param $size
```
Comment (BenMorel) [inline src/Geometry.php]: Missing parameter type

File: src/Geometry.php [hunk 3]
```diff
@@ -526,7 +526,49 @@ public function symDifference(Geometry $geometry)
...
+    {
+        return GeometryEngineRegistry::get()->snapToGrid($this, $size);
+    }
+
+    /**
+     * Returns a "simplified" version of the given geometry using the Douglas-Peucker algorithm.
+     *
+     * @noproxy
+     *
+     * @param $tolerance
```
Comment (BenMorel) [inline src/Geometry.php]: Missing parameter type

File: README.md [hunk 4]
```diff
@@ -120,3 +120,39 @@ Usage
...
+|`locateAlong`  |      |    X    |       |     X      |                  |
+|`locateBetween`|      |    X    |       |     X      |                  |
+|`maxDistance`  |      |    X    |       |     X      |                  |
+|`overlaps`     |  X   |    X    |   X   |     X      |        X         |
+|`relate`       |  X   |    X    |       |     X      |        X         |
+|`simplify`     |  X   |    X    |       |     X      |                  |
+|`snapToGrid`   |      |    X    |       |     X      |                  |
+|`symDifference`|  X   |    X    |       |     X      |        X         |
+|`touches`      |  X   |    X    |   X   |     X      |        X         |
+|`union`        |  X   |    X    |       |     X      |        X         |
```
Comment (BenMorel) [inline README.md]: MySQL does support [union](http://dev.mysql.com/doc/refman/5.6/en/spatial-operator-functions.html#function_st-union)

File: README.md [hunk 5]
```diff
@@ -120,3 +120,39 @@ Usage
...
+--------------------------
+
+This is a list of all functions which are are currently implemented in the geo project. Some functions are only available
+if you use a specific geometry engine.
+
+| Function Name | GEOS | PostGIS | MySQL | SpatiaLite | OpenGIS standard |
+|---------------|------|---------|-------|------------|------------------|
+|`area`         |  X   |    X    |   X   |     X      |        X         |
+|`boundary`     |  X   |    X    |       |     X      |        X         |
+|`buffer`       |  X   |    X    |       |     X      |        X         |
```
Comment (BenMorel) [inline README.md]: MySQL does support [buffer](http://dev.mysql.com/doc/refman/5.6/en/spatial-operator-functions.html#function_buffer)

File: README.md [hunk 6]
```diff
@@ -120,3 +120,39 @@ Usage
...
+|`centroid`     |  X   |    X    |   X   |     X      |        X         |
+|`contains`     |  X   |    X    |   X   |     X      |        X         |
+|`convexHull`   |  X   |    X    |       |     X      |        X         |
+|`crosses`      |  X   |    X    |   X   |     X      |        X         |
+|`difference`   |  X   |    X    |       |     X      |        X         |
```
Comment (BenMorel) [inline README.md]: MySQL does support [difference](http://dev.mysql.com/doc/refman/5.6/en/spatial-operator-functions.html#function_st-difference)

File: README.md [hunk 7]
```diff
@@ -120,3 +120,39 @@ Usage
...
+|`disjoint`     |  X   |    X    |   X   |     X      |        X         |
+|`distance`     |  X   |    X    |       |     X      |        X         |
+|`envelope`     |  X   |    X    |   X   |     X      |        X         |
+|`equals`       |  X   |    X    |   X   |     X      |        X         |
+|`intersects`   |  X   |    X    |   X   |     X      |        X         |
+|`intersection` |  X   |    X    |       |     X      |        X         |
```
Comment (BenMorel) [inline README.md]: MySQL does support [intersection](http://dev.mysql.com/doc/refman/5.6/en/spatial-operator-functions.html#function_st-intersection)

File: README.md [hunk 8]
```diff
@@ -120,3 +120,39 @@ Usage
...
+|`isSimple`     |  X   |    X    |   X   |     X      |        X         |
+|`length`       |  X   |    X    |       |     X      |        X         |
+|`locateAlong`  |      |    X    |       |     X      |                  |
+|`locateBetween`|      |    X    |       |     X      |                  |
+|`maxDistance`  |      |    X    |       |     X      |                  |
+|`overlaps`     |  X   |    X    |   X   |     X      |        X         |
+|`relate`       |  X   |    X    |       |     X      |        X         |
+|`simplify`     |  X   |    X    |       |     X      |                  |
+|`snapToGrid`   |      |    X    |       |     X      |                  |
+|`symDifference`|  X   |    X    |       |     X      |        X         |
```
Comment (BenMorel) [inline README.md]: MySQL does support [symDifference](http://dev.mysql.com/doc/refman/5.6/en/spatial-operator-functions.html#function_st-symdifference)

File: README.md (hunk 4 above)
Comment (cevou) [inline README.md]: Damn, I didn't look in the latest docs ;)

File: README.md (hunk 5 above)
Comment (cevou) [inline README.md]: There is a function `buffer` but not a function `ST_Buffer`. We would have to differentiate between PostGIS and MySQL then.

Comment (BenMorel) [inline README.md]: Just tested it, and `ST_Buffer()` is available as well, it's just not documented yet as far as I can see, so let's include it here!

Comment (cevou) [inline README.md]: OK, I already added it with the last commit. List should be complete now.

**Comment by BenMorel** [comment]
Great job. Thank you!

File: README.md [hunk 9]
```diff
@@ -120,3 +120,39 @@ Usage
...
+Spatial Function Reference
+--------------------------
+
+This is a list of all functions which are are currently implemented in the geo project. Some functions are only available
+if you use a specific geometry engine.
+
+| Function Name | GEOS | PostGIS | MySQL | SpatiaLite | OpenGIS standard |
+|---------------|------|---------|-------|------------|------------------|
+|`area`         |  X   |    X    |   X   |     X      |        X         |
+|`boundary`     |  X   |    X    |   X   |     X      |        X         |
```
Comment (BenMorel) [inline README.md]: Just realized that `ST_Boundary()` seems unsupported by MySQL, did you see it somewhere @cevou ?

Comment (cevou) [inline README.md]: I just had a look again, and didn't find it either.

Comment (BenMorel) [inline README.md]: No worries, I'll just remove the check here, thanks!

**Pull request merged**